controllers/operatorconfig_controller_helper.go

```diff
@@ -31,71 +27,70 @@ func getResources(path string) ([]string, error) {
 	return strings.Split(string(data), "---"), nil
 }
 
-func createResourcesFromFile(path string, ctx context.Context, log logr.Logger) error {
+func (r *OperatorConfigReconciler)createResourcesFromFile(path string, ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
 
 	var decoder = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
-	var dynamicResource dynamic.ResourceInterface
 	errs := []error{}
 
 	data, err := getResources(path)
 	if err != nil {
 		return err
 	}
 
-	cfg, err := rest.InClusterConfig()
-	if err != nil {
-		return err
+	for _, resource := range data {
+		if len(resource) > 0 {
+			obj := &unstructured.Unstructured{}
+			_, _, err := decoder.Decode([]byte(resource), nil, obj)
+			if err != nil {
+				return err
+			}
+
+			err = controllerutil.SetControllerReference(operatorConfig, obj, r.Scheme)
+			if err != nil {
+				errs = append(errs, err)
+			}
+
+			if err = r.Get(ctx, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()},obj); err != nil {
+				if errors.IsNotFound(err) {
+					log.Info("Creating resource", "Kind", obj.GetKind(), "Namespace", obj.GetNamespace(), "Name", obj.GetName())
+					if err := r.Create(ctx, obj); err != nil {
```

Comment:
can we reduce the number if nested ifs here?

1. attempt to create.
2. if error != nil and error != already exists error, return the error, just 1 nesting level
3. if ! already exists error return 1 nesting level
4. assume that if we have reached here it already exists, get the object
5. compare the object with what is in the file
6. if diff found in the above, update

Reply:
may be we can put all the above steps in a separate function so you can iterate cleanly in the loop in line 40. https://github.com/konveyor/mtrho-operator/pull/9/files#diff-4ae62959fd2aa8bb95a07eea198e4ef15603a90010cce3f2f7bd981647f84c04R40